### PR TITLE
Check for data-interval on the first slide of carousel - v4

### LINKED ISSUE #31716: Carousel data-interval is always 5000 on first item
Hi all, I implemented a carousel using bootstrap carousel component. Everything were working fine until I tried to delay the first item with more than 5 seconds. It looks like bootstrap is ignoring `data-interval="10000"` attribute to the element. I checked it in the documentation and it's not working on the demo there as well. First item is delayed only 5 seconds, instead of 10seconds. Every other item is respecting the data-interval attribute. Can you suggest some workaround or am I missing something?

Link to the documentation: https://getbootstrap.com/docs/4.3/components/carousel/#individual-carousel-item-interval
MacOS - Chrome

### PR DESCRIPTION
**Pull request by mitchellfyi**
When starting a cycle for a carousel, it only checks for a default interval, and not an interval defined on the slide element via data props. This adds a check in before creating the interval to move to the next slide.

Fixes #31716

### CODE REVIEW

File: js/tests/unit/carousel.js [hunk 1]
```diff
@@ -173,38 +173,38 @@ $(function () {
...
     var carouselHTML = '<div id="myCarousel" class="carousel slide">' +
-        '<div class="carousel-inner">' +
-        '<div class="carousel-item active">' +
-        '<img alt="">' +
-        '<div class="carousel-caption">' +
-        '<h4>First Thumbnail label</h4>' +
-        '<p>Cras justo odio, dapibus ac facilisis in, egestas eget quam. Donec ' +
-        'id elit non mi porta gravida at eget metus. Nullam id dolor id nibh ' +
-        'ultricies vehicula ut id elit.</p>' +
-        '</div>' +
```
Comment (XhmikosR) [inline js/tests/unit/carousel.js:184]: Please revert all the unrelated changes

**Comment by mitchellbryson** [comment]
Waiting on feedback from V5 PR: https://github.com/twbs/bootstrap/pull/31818

File: js/src/carousel.js [hunk 2]
```diff
@@ -492,6 +503,9 @@ class Carousel {
       $(this._element).trigger(slidEvent)
     }
 
+    // does not wait for the transition to complete
+    this._activeElement = nextElement
```
Comment (Johann-S) [inline js/src/carousel.js:507]: Can you explain this change please?

Comment (mitchellbryson) [reply js/src/carousel.js:507]: this PR is currently out of data (so I reverted it to draft). See https://github.com/twbs/bootstrap/pull/31818

**Comment by XhmikosR** [comment]
@mitchellbryson can you bring this up to par with your main PR and reopen it please?

**Comment by XhmikosR** [comment]
@mitchellbryson friendly ping :) This is currently the only thing pending in our [4.6.0 project](https://github.com/twbs/bootstrap/projects/30)

**Comment by mitchellbryson** [comment]
@XhmikosR sorry for the delay, I'll get on it later this week.

**Comment by XhmikosR** [comment]
Thanks @mitchellbryson, I really appreciate your help and sticking with us ❤

I'm going to merge this since I tested manually and it seems to do the job (and the diff LGTM anyway).

**Review by XhmikosR — APPROVED** [review]

**Comment by mitchellbryson** [comment]
no worries @XhmikosR, thanks for your help getting it through!
